x-pack/plugins/monitoring/server/debugLogger.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import fs from 'fs';
+import { MonitoringConfig } from './config';
+import { RouteDependencies } from './types';
+
+export function decorateDebugServer(
+  _server: any,
+  config: MonitoringConfig,
+  logger: RouteDependencies['logger']
+) {
+  // bail if the proper config value is not set (extra protection)
+  if (!config.ui.debug_mode) {
+    return _server;
+  }
```

Comment:
Check 2 of 2 to make sure we ONLY screw around with this server object if debug_mode is turned on...